src/main/resources/templates/index.ftl

```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html>
+<body>
+<h3>Welcome!</h3>
+<a href="http://localhost:8080/blog/search/">Search blog</a>
+<a href="http://localhost:8080/add/">Add blog</a>
+<#if blogs?has_content>
+        <#list blogs as blog>
+            <fieldset>
+                <p>id: ${blog.getId()}</p>
+                <p>title: ${blog.getTitle()}</p>
+                <p>content: ${blog.getContent()}</p>
+                <form action="http://localhost:8080/update/${blog.getId()}">
+                    <input type="submit" value="Update" />
```

Comment:
In this case and the case below you don't need to use form+action+submit, just add links with href="http://localhost:8080/update/${blog.getId()}" , you don't really do any submission, you just navigate to other pages